File: src/models/bikeListing.ts
import type { BikeType } from './bikeType';

export class BikeListing {
  id?: string;
  title?: string;
  make?: string;
  model?: string;
  year?: number;
  price?: number;
  cc?: number;
  type?: BikeType;
  sellerName?: string;
  location?: string;
  likeCount?: number;
  rating?: number;
  ratingCount?: number;
  phone?: string;
  imageUrl?: string;
  images?: string[];
  mileage?: string;
  km?: string;
  vin?: string;
  description?: string;
  favoritesCount?: number;
}

File: src/api/Entities/BikeListing.cs
using System.ComponentModel.DataAnnotations;

namespace BikerHub.Entities;

public class BikeListing : BaseEntity<Guid>
{

    [MaxLength(100)]
    public string? Make { get; set; }

    [MaxLength(100)]
    public string? Model { get; set; }

    [MaxLength(50)]
    public string? Edition { get; set; }

    public int? Year { get; set; }
    public decimal? Price { get; set; }
    public string? Cc { get; set; }
    public string? Type { get; set; }
    public string? SellerId { get; set; }
    public string? SellerName { get; set; }
    public string? Location { get; set; }
    public double? Rating { get; set; }
    public int RatingCount { get; set; }
    public string? Phone { get; set; }
    public string? Mileage { get; set; }
    public string? Km { get; set; }
    public string? Vin { get; set; }
    public string? Description { get; set; }
    public int FavoritesCount { get; set; }
    public bool IsFavorite { get; set; }
    public bool IsLiked { get; set; }
    public int LikeCount { get; set; }
    public int ViewCount { get; set; }
}
